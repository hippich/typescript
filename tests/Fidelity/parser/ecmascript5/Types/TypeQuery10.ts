var v: typeof A.var;