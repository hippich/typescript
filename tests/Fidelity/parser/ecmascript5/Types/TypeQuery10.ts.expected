{
    "isDeclaration": false,
    "languageVersion": "EcmaScript5",
    "sourceUnit": {
        "kind": "SourceUnit",
        "fullStart": 0,
        "fullEnd": 20,
        "start": 0,
        "end": 20,
        "fullWidth": 20,
        "width": 20,
        "moduleElements": [
            {
                "kind": "VariableStatement",
                "fullStart": 0,
                "fullEnd": 20,
                "start": 0,
                "end": 20,
                "fullWidth": 20,
                "width": 20,
                "modifiers": [],
                "variableDeclaration": {
                    "kind": "VariableDeclaration",
                    "fullStart": 0,
                    "fullEnd": 19,
                    "start": 0,
                    "end": 19,
                    "fullWidth": 19,
                    "width": 19,
                    "varKeyword": {
                        "kind": "VarKeyword",
                        "fullStart": 0,
                        "fullEnd": 4,
                        "start": 0,
                        "end": 3,
                        "fullWidth": 4,
                        "width": 3,
                        "text": "var",
                        "value": "var",
                        "valueText": "var",
                        "hasTrailingTrivia": true,
                        "trailingTrivia": [
                            {
                                "kind": "WhitespaceTrivia",
                                "fullStart": 3,
                                "fullEnd": 4,
                                "text": " "
                            }
                        ]
                    },
                    "variableDeclarators": [
                        {
                            "kind": "VariableDeclarator",
                            "fullStart": 4,
                            "fullEnd": 19,
                            "start": 4,
                            "end": 19,
                            "fullWidth": 15,
                            "width": 15,
                            "propertyName": {
                                "kind": "IdentifierName",
                                "fullStart": 4,
                                "fullEnd": 5,
                                "start": 4,
                                "end": 5,
                                "fullWidth": 1,
                                "width": 1,
                                "text": "v",
                                "value": "v",
                                "valueText": "v"
                            },
                            "typeAnnotation": {
                                "kind": "TypeAnnotation",
                                "fullStart": 5,
                                "fullEnd": 19,
                                "start": 5,
                                "end": 19,
                                "fullWidth": 14,
                                "width": 14,
                                "colonToken": {
                                    "kind": "ColonToken",
                                    "fullStart": 5,
                                    "fullEnd": 7,
                                    "start": 5,
                                    "end": 6,
                                    "fullWidth": 2,
                                    "width": 1,
                                    "text": ":",
                                    "value": ":",
                                    "valueText": ":",
                                    "hasTrailingTrivia": true,
                                    "trailingTrivia": [
                                        {
                                            "kind": "WhitespaceTrivia",
                                            "fullStart": 6,
                                            "fullEnd": 7,
                                            "text": " "
                                        }
                                    ]
                                },
                                "type": {
                                    "kind": "TypeQuery",
                                    "fullStart": 7,
                                    "fullEnd": 19,
                                    "start": 7,
                                    "end": 19,
                                    "fullWidth": 12,
                                    "width": 12,
                                    "typeOfKeyword": {
                                        "kind": "TypeOfKeyword",
                                        "fullStart": 7,
                                        "fullEnd": 14,
                                        "start": 7,
                                        "end": 13,
                                        "fullWidth": 7,
                                        "width": 6,
                                        "text": "typeof",
                                        "value": "typeof",
                                        "valueText": "typeof",
                                        "hasTrailingTrivia": true,
                                        "trailingTrivia": [
                                            {
                                                "kind": "WhitespaceTrivia",
                                                "fullStart": 13,
                                                "fullEnd": 14,
                                                "text": " "
                                            }
                                        ]
                                    },
                                    "name": {
                                        "kind": "QualifiedName",
                                        "fullStart": 14,
                                        "fullEnd": 19,
                                        "start": 14,
                                        "end": 19,
                                        "fullWidth": 5,
                                        "width": 5,
                                        "left": {
                                            "kind": "IdentifierName",
                                            "fullStart": 14,
                                            "fullEnd": 15,
                                            "start": 14,
                                            "end": 15,
                                            "fullWidth": 1,
                                            "width": 1,
                                            "text": "A",
                                            "value": "A",
                                            "valueText": "A"
                                        },
                                        "dotToken": {
                                            "kind": "DotToken",
                                            "fullStart": 15,
                                            "fullEnd": 16,
                                            "start": 15,
                                            "end": 16,
                                            "fullWidth": 1,
                                            "width": 1,
                                            "text": ".",
                                            "value": ".",
                                            "valueText": "."
                                        },
                                        "right": {
                                            "kind": "IdentifierName",
                                            "fullStart": 16,
                                            "fullEnd": 19,
                                            "start": 16,
                                            "end": 19,
                                            "fullWidth": 3,
                                            "width": 3,
                                            "text": "var",
                                            "value": "var",
                                            "valueText": "var",
                                            "isKeywordConvertedToIdentifier": true
                                        }
                                    }
                                }
                            }
                        }
                    ]
                },
                "semicolonToken": {
                    "kind": "SemicolonToken",
                    "fullStart": 19,
                    "fullEnd": 20,
                    "start": 19,
                    "end": 20,
                    "fullWidth": 1,
                    "width": 1,
                    "text": ";",
                    "value": ";",
                    "valueText": ";"
                }
            }
        ],
        "endOfFileToken": {
            "kind": "EndOfFileToken",
            "fullStart": 20,
            "fullEnd": 20,
            "start": 20,
            "end": 20,
            "fullWidth": 0,
            "width": 0,
            "text": ""
        }
    },
    "lineMap": {
        "lineStarts": [
            0
        ],
        "length": 20
    }
}